src/hooks/balances/lib/getBalancesByAddress.tsx

```diff
@@ -0,0 +1,87 @@
+import { ApiPromise } from '@polkadot/api';
+import { includes } from 'lodash';
+import { Balance } from '../../../generated/graphql';
+import constants from '../../../constants';
+import { OrmlAccountData } from '@open-web3/orml-types/interfaces';
+import '@polkadot/api-augment';
+import { FrameSystemAccountInfo } from '@polkadot/types/lookup';
+
+// this function fetches asset balances only for a given set of assetIds
+export const getBalancesByAddress = async (
+  apiInstance: ApiPromise,
+  address: string,
+  assetIds: string[]
+): Promise<Balance[]> => {
+  let balances: Balance[] = [];
+
+  if (includes(assetIds, constants.nativeAssetId)) {
+    const nativeBalance = await fetchNativeAssetBalance(apiInstance, address);
+    balances.push(nativeBalance);
+  }
+
+  // make sure that there is no native assetId
+  const nonNativeAssetIds = assetIds.filter(
+    (id) => id !== constants.nativeAssetId
+  );
+  // fetch non-native assets only if needed
+  if (nonNativeAssetIds.length) {
+    const nonNativeBalances = await fetchNonNativeAssetBalances(
+      apiInstance,
+      address,
+      nonNativeAssetIds
+    );
+
+    balances.push(...nonNativeBalances);
+  }
+
+  return balances;
+};
+
+export const fetchNativeAssetBalance = async (
+  apiInstance: ApiPromise,
+  address: string
+): Promise<Balance> => {
+  // no handling of undefined because apiInstance returns default value of 0 for native asset
+  const nativeAssetBalance: FrameSystemAccountInfo =
```

Comment:
Typing here is probably a little excessive, since the `system.account` call is well typed already, but it works.

Reply:
This was the place where the type augmenting broke with the polkadot js upgrade to v7. I can remove it.